### PR TITLE
multiclient's validator import

### PR DESCRIPTION
**Pull request by gbayasgalan**
https://github.com/stereum-dev/ethereum2-control-center-web/issues/76
- also multiclient's config-export & import (without validators)
- other clients import-config with validators (validators not tested yet) 
- Age encryption tool is used to export-and import configuration file ( instead GPG ) - https://github.com/stereum-dev/sba-audit-2021/issues/3
- importing configuration with Validators for all Clients

### CODE REVIEW

File: roles/stop-services/tasks/main.yaml [hunk 1]
```diff
@@ -5,3 +5,10 @@
     chdir: "{{ e2dc_install_path }}"
   become: yes
   become_user: "{{ stereum_user }}"
+  when: setup != "multiclient"
+
+- name: Stop services
```
Comment (stefa2k) [inline roles/stop-services/tasks/main.yaml:10]: this is doing the same as the other task `docker-compose stop`

File: roles/start-services/tasks/main.yaml [hunk 2]
```diff
@@ -5,3 +5,10 @@
     chdir: "{{ e2dc_install_path }}"
   become: yes
   become_user: "{{ stereum_user }}"
+  when: setup != "multiclient"
+
+- name: Start services multiclient
```
Comment (stefa2k) [inline roles/start-services/tasks/main.yaml:10]: this is doing the same as the task above

File: roles/import-validator-keys-multiclient/tasks/main.yaml [hunk 3]
```diff
@@ -29,14 +79,13 @@
...
   args:
-    executable: /bin/bash
+    executable: /bin/bash  
+  become: yes 
 
 - name: Copy wallet into Stereum's wallets directory
-  command: cp -r ~/.config/ethereum2/wallets/. {{ e2dc_install_path }}/wallets
-  become: yes
-
-- name: Set permissions of wallet directory
```
Comment (stefa2k) [inline roles/import-validator-keys-multiclient/tasks/main.yaml:38]: Is this not necessary anymore even with non-root-user access?

File: roles/import-validator-keys-multiclient/tasks/main.yaml [hunk 4]
```diff
@@ -8,10 +8,60 @@
...
+  when: go_check.stat.exists and go_check.stat.isdir
+
+- name: Remove Go directory if exists
+  ansible.builtin.file:
+    path: "{{ homedir.stdout }}/go"
+    state: absent
+  when: go_check.stat.exists and go_check.stat.isdir
+
+- name: Download and Extract Go binary in '$HOME'
+  shell: wget -c https://dl.google.com/go/go1.16.8.linux-amd64.tar.gz -O - | sudo tar -xz -C "{{ homedir.stdout }}"
```
Comment (stefa2k) [inline roles/import-validator-keys-multiclient/tasks/main.yaml:42]: why not use `apt install golang` or `yum install golang`?

File: roles/check-keys-password-teku/tasks/main.yaml [hunk 5]
```diff
@@ -43,6 +43,16 @@
...
 
+- name: Remove keystores files from launchpad 'import-config'
+  shell: rm "{{ e2dc_install_path }}/launchpad/*.json"
+  become: yes
+  when: exported_config_path is defined and (validator_log.stdout is search('Failed to decrypt keystore') or validator_log.stdout is search('Keystore password cannot be empty'))
+
+- name: Remove password files from launchpad 'import-config'
+  shell: rm "{{ e2dc_install_path }}/launchpad/*.txt"
+  become: yes
+  when: exported_config_path is defined and (validator_log.stdout is search('Failed to decrypt keystore') or validator_log.stdout is search('Keystore password cannot be empty'))
```
Comment (stefa2k) [inline roles/check-keys-password-teku/tasks/main.yaml:54]: please use parantheses to make it clear when `and` and `or` is used

**Review by stefa2k — CHANGES_REQUESTED** [review]

File: roles/stop-services/tasks/main.yaml (hunk 1 above)
Comment (gbayasgalan) [reply roles/stop-services/tasks/main.yaml:10]: got it

File: roles/start-services/tasks/main.yaml (hunk 2 above)
Comment (gbayasgalan) [reply roles/start-services/tasks/main.yaml:10]: got it

File: roles/import-validator-keys-multiclient/tasks/main.yaml (hunk 4 above)
Comment (gbayasgalan) [reply roles/import-validator-keys-multiclient/tasks/main.yaml:42]: Ethdo will use Go version 1.6... while `apt install golang` & `yum install golang` will install Go version1.3... .

File: export-config.yaml [hunk 6]
```diff
@@ -4,12 +4,48 @@
...
       become: yes
 
+    - name: Create directory for exporting wallet and passphrases
+      ansible.builtin.file:
+        path: /tmp/exported-config/wallet
+        state: directory
+      become: yes  
+      when: setup == "multiclient"
+
+    - name: Install pip
```
Comment (stefa2k) [inline export-config.yaml:39]: I think there needs to be a `become: yes`

File: export-config.yaml [hunk 7]
```diff
@@ -4,12 +4,48 @@
...
+      apt:
+        update_cache: yes
+        name:
+          - python3-pip
+
+    - name: Install pexpect
```
Comment (stefa2k) [inline export-config.yaml:45]: `become: yes`

File: import-validator-accounts.yaml [hunk 8]
```diff
@@ -80,6 +70,12 @@
     - name: Checking keys' password of teku
       include_role:
         name: check-keys-password-teku
-      when: setup == "teku"
+      when: setup == "teku" 
+
+    - name: Remove exported-config directory
```
Comment (stefa2k) [inline import-validator-accounts.yaml:75]: maybe `become: yes` needed?

File: roles/check-keys-password-teku/tasks/main.yaml [hunk 9]
```diff
@@ -43,6 +43,16 @@
   with_items: '{{importing_key_cli.stdout_lines}}'
   when: validator_keys is not defined and (validator_log.stdout is search('Failed to decrypt keystore') or validator_log.stdout is search('Keystore password cannot be empty'))
 
+- name: Remove keystores files from launchpad 'import-config'
+  shell: rm "{{ e2dc_install_path }}/launchpad/*.json"
```
Comment (stefa2k) [inline roles/check-keys-password-teku/tasks/main.yaml:47]: you can use `file:` with `state: absent` https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_module.html#parameter-state

File: roles/check-keys-password-teku/tasks/main.yaml [hunk 10]
```diff
@@ -43,6 +43,16 @@
...
+  become: yes
+  when: exported_config_path is defined and (validator_log.stdout is search('Failed to decrypt keystore') or validator_log.stdout is search('Keystore password cannot be empty'))
+
+- name: Remove password files from launchpad 'import-config'
+  shell: rm "{{ e2dc_install_path }}/launchpad/*.txt"
```
Comment (stefa2k) [inline roles/check-keys-password-teku/tasks/main.yaml:52]: you can use `file:` with `state: absent` https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_module.html#parameter-state

File: roles/import-validator-keys-multiclient/tasks/main.yaml [hunk 11]
```diff
@@ -8,17 +8,76 @@
     - git
     - build-essential
     - gcc
-    - golang
+    - wget
+    - python3-pip  
+
+- name: Install pexpect
```
Comment (stefa2k) [inline roles/import-validator-keys-multiclient/tasks/main.yaml:14]: I think `become: yes` is needed here in case user doesn't use root

**Review by stefa2k — CHANGES_REQUESTED** [review]

**Pull request merged**